android/java/org/chromium/chrome/browser/omnibox/BraveLocationBarMediator.java

```diff
@@ -0,0 +1,151 @@
+            @NonNull LocationBarDataProvider locationBarDataProvider,
+            @NonNull ObservableSupplier<Profile> profileSupplier,
+            @NonNull PrivacyPreferencesManager privacyPreferencesManager,
+            @NonNull OverrideUrlLoadingDelegate overrideUrlLoadingDelegate,
+            @NonNull LocaleManager localeManager,
+            @NonNull OneshotSupplier<TemplateUrlService> templateUrlServiceSupplier,
+            @NonNull BackKeyBehaviorDelegate backKeyBehavior, @NonNull WindowAndroid windowAndroid,
+            boolean isTablet, @NonNull SearchEngineLogoUtils searchEngineLogoUtils,
+            @NonNull LensController lensController,
+            @NonNull Runnable launchAssistanceSettingsAction,
+            @NonNull SaveOfflineButtonState saveOfflineButtonState, @NonNull OmniboxUma omniboxUma,
+            @NonNull BooleanSupplier isToolbarMicEnabledSupplier) {
+        super(context, locationBarLayout, locationBarDataProvider, profileSupplier,
+                privacyPreferencesManager, overrideUrlLoadingDelegate, localeManager,
+                templateUrlServiceSupplier, backKeyBehavior, windowAndroid, isTablet,
+                searchEngineLogoUtils, lensController, launchAssistanceSettingsAction,
+                saveOfflineButtonState, omniboxUma, isToolbarMicEnabledSupplier);
+    }
+
+    @Override
+    void updateButtonVisibility() {
+        super.updateButtonVisibility();
+        updateQRButtonVisibility();
+    }
+
+    @Override
+    public void onPrimaryColorChanged() {
+        super.onPrimaryColorChanged();
+        updateQRButtonColors();
+    }
+
+    @Override
+    public void onAssistantVoiceSearchServiceChanged() {
+        super.onAssistantVoiceSearchServiceChanged();
+        updateQRButtonColors();
+    }
+
+    void updateQRButtonColors() {
+        AssistantVoiceSearchService assistantVoiceSearchService =
+                mAssistantVoiceSearchServiceSupplier.get();
+        if (assistantVoiceSearchService == null) return;
+
+        if (mLocationBarLayout instanceof BraveLocationBarLayout) {
+            ((BraveLocationBarLayout) mLocationBarLayout)
+                    .setQRButtonTint(assistantVoiceSearchService.getButtonColorStateList(
+                            mBrandedColorScheme, mContext));
+        }
+    }
+
+    private void updateQRButtonVisibility() {
+        if (mLocationBarLayout instanceof BraveLocationBarLayout) {
+            ((BraveLocationBarLayout) mLocationBarLayout)
+                    .setQRButtonVisibility(shouldShowQRButton());
+        }
+    }
+
+    private boolean shouldShowQRButton() {
+        if (!mNativeInitialized) {
+            return false;
+        }
+        if (mIsTablet) {
+            return mUrlHasFocus || mIsUrlFocusChangeInProgress;
+        } else {
+            boolean deleteButtonVisible = (boolean) BraveReflectionUtil.InvokeMethod(
+                    LocationBarMediator.class, this, "shouldShowDeleteButton");
```

Comment:
We should avoid `BraveReflectionUtil.InvokeMethod` where possible, as it forces us to add additional proguard exceptions. In this case it's better to do it via `makePublicMethod` capability.

Reply:
sure, updating now